tests/unit/source_code/linters/test_dfsa.py

```diff
@@ -0,0 +1,149 @@
+        ("s3a://bucket1/folder1", True),
+        ("/dbfs/mnt/foo/bar", True),
+        ("/tmp/foo", False),
+        ("table.we.know.nothing.about", False),
+    ],
+)
+def test_matches_dfsa_pattern(path, matches):
+    """see https://github.com/databrickslabs/ucx/issues/2350"""
+    matched = any(pattern.matches(path) for pattern in DFSA_PATTERNS)
+    assert matches == matched
+
+
+@pytest.mark.parametrize(
+    "code, expected",
+    [
+        ('SOME_CONSTANT = "not a file system path"', 0),
+        ('SOME_CONSTANT = ("/dbfs/mnt", "dbfs:/", "/mnt/")', 3),
+        ('# "/dbfs/mnt"', 0),
+        ('SOME_CONSTANT = "/dbfs/mnt"', 1),
+        ('SOME_CONSTANT = "/dbfs/mnt"; load_data(SOME_CONSTANT)', 1),
+        ('SOME_CONSTANT = 42; load_data(SOME_CONSTANT)', 0),
+    ],
+)
+def test_detects_dfsa_paths(code, expected):
+    linter = DfsaPyLinter(CurrentSessionState(), allow_spark_duplicates=True)
+    advices = list(linter.lint(code))
+    for advice in advices:
+        assert isinstance(advice, Advice)
+    assert len(advices) == expected
+
+
+@pytest.mark.parametrize(
+    "code, expected",
+    [
+        ("load_data('/dbfs/mnt/data')", 1),
+        ("load_data('/data')", 1),
+        ("load_data('/dbfs/mnt/data', '/data')", 2),
+        ("# load_data('/dbfs/mnt/data', '/data')", 0),
+        ('spark.read.parquet("/mnt/foo/bar")', 1),
+        ('spark.read.parquet("dbfs:/mnt/foo/bar")', 1),
+        ('spark.read.parquet("dbfs://mnt/foo/bar")', 1),
+        ('DBFS="dbfs:/mnt/foo/bar"; spark.read.parquet(DBFS)', 1),
+        (
+            """
+DBFS1="dbfs:/mnt/foo/bar1"
+systems=[DBFS1, "dbfs:/mnt/foo/bar2"]
+for system in systems:
+    spark.read.parquet(system)
+""",
+            2,
+        ),
+    ],
+)
+def test_dfsa_usage_linter(code, expected):
+    linter = DfsaPyLinter(CurrentSessionState(), allow_spark_duplicates=True)
+    advices = linter.lint(code)
+    count = 0
+    for advice in advices:
+        if isinstance(advice, Deprecation):
+            count += 1
+    assert count == expected
+
+
+def test_dfsa_name():
+    linter = DfsaPyLinter(CurrentSessionState())
+    assert linter.name() == "dfsa-usage"
+
+
+@pytest.mark.parametrize(
+    "query",
+    [
+        "SELECT * FROM old.things LEFT JOIN hive_metastore.other.matters USING (x) WHERE state > 1 LIMIT 10",
+        # Make sure non-sql doesn't just fail
+        "print('hello')",
+        "",
+    ],
+)
+def test_non_dfsa_triggers_nothing(query):
+    ftf = DfsaSqlLinter()
+    assert not list(ftf.lint(query))
+
+
+@pytest.mark.parametrize(
+    "query, table",
+    [
+        ('SELECT * FROM parquet.`dbfs:/...` LIMIT 10', "dbfs:/..."),
+        ("SELECT * FROM delta.`/mnt/...` WHERE foo > 6", "/mnt/..."),
+        ("SELECT * FROM json.`/a/b/c` WHERE foo > 6", "/a/b/c"),
+        ("DELETE FROM json.`/...` WHERE foo = 'bar'", "/..."),
+        (
+            "MERGE INTO delta.`/dbfs/...` t USING source ON t.key = source.key WHEN MATCHED THEN DELETE",
```

Comment:
can you make this line shorter for the entire example fit into one line?

Reply:
done